GooglePlayInstant/Editor/AndroidManifest/AndroidManifestHelper.cs

```diff
@@ -0,0 +1,264 @@
+        internal const string PreconditionOneApplicationElement = "expect 1 application element";
+        internal const string PreconditionOneMainActivity = "expect 1 activity with action MAIN and category LAUNCHER";
+
+        /// <summary>
+        /// Creates a new XDocument representing a basic Unity AndroidManifest XML file.
+        /// </summary>
+        public static XDocument CreateManifestXDocument()
+        {
+            return new XDocument(new XElement(
+                Manifest,
+                new XAttribute(AndroidXmlns, XNamespace.Get(AndroidNamespaceUrl)),
+                new XElement(Application,
+                    new XElement(Activity,
+                        new XAttribute(AndroidNameXName, "com.unity3d.player.UnityPlayerActivity"),
+                        new XElement(IntentFilter,
+                            new XElement(Action, new XAttribute(AndroidNameXName, ActionMain)),
+                            new XElement(Category, new XAttribute(AndroidNameXName, CategoryLauncher)))))));
+        }
+
+        /// <summary>
+        /// Converts the specified XDocument representing an AndroidManifest to support an installed app build.
+        /// </summary>
+        public static void ConvertManifestToInstalled(XDocument doc)
+        {
+            foreach (var manifestElement in doc.Elements(Manifest))
+            {
+                manifestElement.Attributes(AndroidTargetSandboxVersionXName).Remove();
+                foreach (var applicationElement in manifestElement.Elements(Application))
+                {
+                    foreach (var mainActivity in GetMainActivities(applicationElement))
+                    {
+                        // TODO: also remove view intent filters?
+                        GetDefaultUrlMetaDataElements(mainActivity).Remove();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts the specified XDocument representing an AndroidManifest to support an instant app build.
+        /// </summary>
+        /// <param name="doc">An XDocument representing an AndroidManifest.</param>
+        /// <param name="uri">The Default URL to use, or null for a URL-less instant app.</param>
+        /// <returns>An error message if there was a problem updating the manifest, or null if successful.</returns>
+        public static string ConvertManifestToInstant(XDocument doc, Uri uri)
+        {
+            var manifestElement = GetExactlyOne(doc.Elements(Manifest));
+            if (manifestElement == null)
+            {
+                return PreconditionOneManifestElement;
+            }
+
+            var androidAttribute = manifestElement.Attribute(AndroidXmlns);
+            if (androidAttribute == null)
+            {
+                return PreconditionMissingXmlnsAndroid;
+            }
+
+            if (androidAttribute.Value != AndroidNamespaceUrl)
```

Comment:
Shouldn't

    if (androidAttribute != AndroidNamespaceUrl)

automatically unbox the androidAttribute into the underlying string type?

Reply:
Not in this case:
AndroidManifestHelper.cs(116, 17): [CS0019] Operator '!=' cannot be applied to operands of type 'XAttribute' and 'string'